command/node_status.go

```diff
@@ -333,86 +343,109 @@ func (c *NodeStatusCommand) formatNode(client *api.Client, node *api.Node) int {
-			c.Ui.Output(formatList(hostResources))
-		}
+	// Emit node events
+	c.outputNodeStatusEvents(node)
 
-		if err == nil && node.NodeResources != nil && len(node.NodeResources.Devices) > 0 {
-			c.Ui.Output(c.Colorize().Color("\n[bold]Device Resource Utilization[reset]"))
-			c.Ui.Output(formatList(getDeviceResourcesForNode(hostStats.DeviceStats, node)))
-		}
-		if hostStats != nil && c.stats {
-			c.Ui.Output(c.Colorize().Color("\n[bold]CPU Stats[reset]"))
-			c.printCpuStats(hostStats)
-			c.Ui.Output(c.Colorize().Color("\n[bold]Memory Stats[reset]"))
-			c.printMemoryStats(hostStats)
-			c.Ui.Output(c.Colorize().Color("\n[bold]Disk Stats[reset]"))
-			c.printDiskStats(hostStats)
-			if len(hostStats.DeviceStats) > 0 {
-				c.Ui.Output(c.Colorize().Color("\n[bold]Device Stats[reset]"))
-				printDeviceStats(c.Ui, hostStats.DeviceStats)
-			}
-		}
+	// Get list of running allocations on the node
+	runningAllocs, err := getRunningAllocs(client, node.ID)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error querying node for running allocations: %s", err))
+		return 1
 	}
 
-	nodeAllocs, _, err := client.Nodes().Allocations(node.ID, nil)
+	allocatedResources := getAllocatedResources(client, runningAllocs, node)
+	c.Ui.Output(c.Colorize().Color("\n[bold]Allocated Resources[reset]"))
+	c.Ui.Output(formatList(allocatedResources))
+
+	actualResources, err := getActualResources(client, runningAllocs, node)
+	if err == nil {
+		c.Ui.Output(c.Colorize().Color("\n[bold]Allocation Resource Utilization[reset]"))
+		c.Ui.Output(formatList(actualResources))
+	}
+
+	hostResources, err := getHostResources(hostStats, node)
 	if err != nil {
+		c.Ui.Output("")
+		c.Ui.Error(fmt.Sprintf("error fetching node stats: %v", err))
+	}
+	if err == nil {
+		c.Ui.Output(c.Colorize().Color("\n[bold]Host Resource Utilization[reset]"))
+		c.Ui.Output(formatList(hostResources))
+	}
+
+	if err == nil && node.NodeResources != nil && len(node.NodeResources.Devices) > 0 {
+		c.Ui.Output(c.Colorize().Color("\n[bold]Device Resource Utilization[reset]"))
+		c.Ui.Output(formatList(getDeviceResourcesForNode(hostStats.DeviceStats, node)))
+	}
+	if hostStats != nil && c.stats {
+		c.Ui.Output(c.Colorize().Color("\n[bold]CPU Stats[reset]"))
+		c.printCpuStats(hostStats)
+		c.Ui.Output(c.Colorize().Color("\n[bold]Memory Stats[reset]"))
+		c.printMemoryStats(hostStats)
+		c.Ui.Output(c.Colorize().Color("\n[bold]Disk Stats[reset]"))
+		c.printDiskStats(hostStats)
+		if len(hostStats.DeviceStats) > 0 {
+			c.Ui.Output(c.Colorize().Color("\n[bold]Device Stats[reset]"))
+			printDeviceStats(c.Ui, hostStats.DeviceStats)
+		}
+	}
+
+	if err := c.outputAllocInfo(client, node); err != nil {
 		c.Ui.Error(fmt.Sprintf("Error querying node allocations: %s", err))
 		return 1
 	}
 
+	return 0
+}
+
+func (c *NodeStatusCommand) outputAllocInfo(client *api.Client, node *api.Node) error {
+	nodeAllocs, _, err := client.Nodes().Allocations(node.ID, nil)
+	if err != nil {
+		return err
```

Comment:
I'd return the error prefix here instead of the caller side.  Had the function been `queryAllocInfo`, having callers add the prefix would be reasonable imo.

```suggestion
		return fmt.Errorf("failed to query node allocations: %s", err)
```